crates/wysiwyg/src/dom/nodes/dom_node.rs

```diff
@@ -34,6 +36,7 @@ where
     Container(ContainerNode<S>), // E.g. html, div
     Text(TextNode<S>),
     LineBreak(LineBreakNode<S>),
+    Mention(MentionNode<S>),
```

Comment:
For now I left the name of this node as `Mention` to be explicit about what it is. If later there are other nodes that should behave in a similar way, we could think about renaming this to `Object` or similar.